frame/evm/precompile/contracts/kton/src/lib.rs

```diff
@@ -108,14 +110,39 @@ impl<T: Config> Precompile for Kton<T> {
 				Ok((ExitSucceed::Returned, vec![], 20000))
 			}
 			Action::Withdraw(wd) => {
-				let withdraw_value = wd.kton_value.saturating_mul(helper);
-				T::KtonCurrency::transfer(
-					&con_caller,
+				// Ensure wkton is a contract
+				ensure!(
+					!crate::Module::<T>::is_contract_code_empty(&context.caller),
+					ExitError::Other("The caller must be wkton contract!".into())
+				);
+				// Ensure context's apparent_value is zero
+				ensure!(
+					context.apparent_value == U256::zero(),
+					ExitError::Other("The value in tx must be zero!".into())
```

Comment:
The apparent value in tx

Reply:
I don't know what apparent actually means here, probably i'm wrong

Reply:
The `apparent` refers to tx value. here refers to context inner field from EVM executor.